sdk/metric/exemplar/fixed_size_reservoir.go

```diff
@@ -26,7 +27,13 @@ func FixedSizeReservoirProvider(k int) ReservoirProvider {
 // sample each one. If there are more than k, the Reservoir will then randomly
 // sample all additional measurement with a decreasing probability.
 func NewFixedSizeReservoir(k int) *FixedSizeReservoir {
-	return newFixedSizeReservoir(newStorage(k))
+	if k < 0 {
+		k = 0
+	}
```

Comment:
Combining the count and next into a single `uint64` means we can only support reservoir sizes up to `uint32` now. The value of `k` should be clamped here and appropriately updated in all uses.